src/server/api/routers/groups.ts

```diff
@@ -27,9 +28,11 @@ export const groupRouter = createTRPCRouter({
   }),
 
   fetchGroups: publicProcedure.query(async ({ ctx }) => {
```

Comment:
if this should be fetching the logged in user's groups then we want this to be a protected procedure. As is if someone is logged out and this function is called userId will be undefined and this database query will not function as we want.

Reply:
i changed it back to protected and handled a user not being logged in on client side myggroups.tsx

```
if (groupsList.isLoading) {
    return (
      <>
      {commonHead}
        <div>Loading Groups...</div>
      </>
    )
  }
  if (groupsList.isError) {
    setDisplayLoginModal(true)
    return (
      <>
      {commonHead}
        <div>eError: Unable to fetch groups. Please log in.</div>
      </>
    )
  }
  return (
    <>
      {commonHead}
        <div className="flex flex-col gap-4 w-full">
        {groupsList.data?.groups?.map((group) => (
```
etc..